extensions/amp-list/0.1/amp-list.js

```diff
@@ -514,24 +526,35 @@ export class AmpList extends AMP.BaseElement {
       (isFetch && this.element.hasAttribute('reset-on-refresh')) ||
       this.element.getAttribute('reset-on-refresh') === 'always'
     ) {
-      // Placeholder and loading don't need a mutate context.
-      this.togglePlaceholder(true);
-      this.toggleLoading(true, /* opt_force */ true);
-      this.mutateElement(() => {
-        this.toggleFallback_(false);
-        // Clean up bindings in children before removing them from DOM.
-        if (this.bind_) {
-          const removed = toArray(this.container_.children);
-          this.bind_.rescan(/* added */ [], removed, {
-            'fast': true,
-            'update': false,
-          });
-        }
-        removeChildren(dev().assertElement(this.container_));
-        if (this.loadMoreEnabled_) {
-          this.getLoadMoreService_().hideAllLoadMoreElements();
+      let currentHeight;
+      this.measureMutateElement(
+        () => {
+          currentHeight = this.element./*OK*/ offsetHeight;
+        },
+        () => {
```

Comment:
Can we de-dupe this into `lockHeight` and `unlockHeight` helper methods?

Reply:
Is it awkward to put the measure/offsetHeight into the helper methods?

Reply:
You mean:
```js
return this.maybeLockHeight().then(() => this.mutateElement(() => { /* common stuff */ });
```
with
```js
maybeLockHeight() {
  if (!notContainer) {
    return Promise.resolve();
  }
  return measureMutateElement(() => { /* get offset */ }, () => { /* set styles */ });
}
```

Alternatively:
```js
const commonStuff = () => this.mutateElement(() => { /* common stuff */ });
if (container) {
  return this.lockHeight().then(() => commonStuff());
} 
return commonStuff();
```

Are the two consecutive mutate blocks consolidated by the runtime at all? Also this will make the unit tests become more spaghetti.

Reply:
Yup, that's what I mean. You could also massage it a bit more if you want to avoid extra mutate calls.